pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -102,69 +101,64 @@ func (p RESTStorageProvider) GroupName() string {
 // PostStartHook returns the hook func that launches the config provider
 func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
 	return PostStartHookName, func(hookContext genericapiserver.PostStartHookContext) error {
-		flowcontrolClientSet := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+		clientset := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+
 		go func() {
-			const retryCreatingSuggestedSettingsInterval = time.Second
-			err := wait.PollImmediateUntil(
-				retryCreatingSuggestedSettingsInterval,
+			if err := wait.PollImmediateUntil(
```

Comment:
I think the post-start-hook shouldn't finish until this first round actually is finished (I don't want apiserver to start before e.g. mandatory configuration is created).

So I think that the first loop should be outside the go-routine.

Reply:
Friendly ping

Reply:
@wojtek-t today apf loads the mandatory configuration objects from memory when it starts so it can operate with minimal configuration and this is before the apiserver is ready to accept request I think. 

Failing the post startup hook in this situation seems more drastic while we can still have the apiserver keep functioning with minimal apf configuration and allow the cluster operator to debug the issue. Thoughts?

On the other hand, the question is do we need the mandatory configurations as objects on cluster, An alternate is to  always load them from memory.  

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go#L200
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go#L482-L488

```
	// Supply missing mandatory PriorityLevelConfiguration objects
	if !meal.haveExemptPL {
		meal.imaginePL(fcboot.MandatoryPriorityLevelConfigurationExempt, cfgCtlr.requestWaitLimit)
	}
	if !meal.haveCatchAllPL {
		meal.imaginePL(fcboot.MandatoryPriorityLevelConfigurationCatchAll, cfgCtlr.requestWaitLimit)
	}
```

Reply:
> Failing the post startup hook in this situation seems more drastic while we can still have the apiserver keep functioning with minimal apf configuration and allow the cluster operator to debug the issue. Thoughts?

Technically we can, but it's much more fragile. The question is do we really want to.
There are two cases:
- either there are some temporary errors which prevents us from reconciliation, and then waiting a bit before making kube-apiserver ready doesn't hurt much
- or those are permanent, and then it will never reconcile, and then not making kube-apiserver not ready would be a good signal to operator that something is wrong and we will never reconcile
In both cases, blocking sounds like a desired option to me.
@deads2k @lavalamp - thoughts?

Reply:
Given a fast first application of these resources (which I would expect), I think it is reasonable to require these to be written at least once before completing the poststarthook.

I do think it is easier to understand the mechanics by having these exposed for visibility in the API.